Makefile

```diff
@@ -106,7 +106,7 @@ help: ## Display this help.
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) \
 		rbac:roleName=manager-role output:rbac:artifacts:config=config/components/rbac\
-		crd output:crd:artifacts:config=config/components/crd/bases\
+		crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/components/crd/bases\
```

Comment:
This is needed to generate the `metadata.labels` and `metadata.annotations` in the CRD schema for the template (`PodTemplateSpec) in PodSet. Without it, the fields are dropped at the client side. 

The template fields are used to keep the original labels, annotations, nodeSelector, and tolerations. 

`generateEmbeddedObjectMeta` is also used for other projects such as:
- [jobset](https://github.com/kubernetes-sigs/jobset/blob/68d1afb37c1b102d1f69c6710c5c42081c49181c/Makefile#L85)
- [mpioperator](https://github.com/kubeflow/mpi-operator/blob/64db27bef2bc34bc1c9af3112a8452dcb2dd9bb2/Makefile#L46)
- [training-operator](https://github.com/kubeflow/training-operator/blob/a4c0cec561a4bfe478720f1a102f305ed656071b/Makefile#L4)

Alternatively, I could keep labels and annotation directly on newly declared fields in `PodSet`, but it seems preferred to avoid new fields.

Reply:
Thanks